src/core/thread/thread_netif.cpp

```diff
@@ -94,18 +94,16 @@ const char *ThreadNetif::GetName(void) const
 
 ThreadError ThreadNetif::Up(void)
 {
-    ThreadError error = kThreadError_None;
-
-    VerifyOrExit(!mIsUp, error = kThreadError_Already);
-
-    mIp6.AddNetif(*this);
-    mMeshForwarder.Start();
-    mCoapServer.Start();
-    mMleRouter.Enable();
-    mIsUp = true;
+    if (!mIsUp)
```

Comment:
I disagree. I am a huge advocate of the original [`require` macro design pattern](http://www.mactech.com/articles/develop/issue_11/Parent_final.html), I don't see any utility (aesthetic or otherwise) for using that pattern here. Please elaborate if I am missing something.